### ISSUE TITLE
HOT RELOADING feature not working as expected!!

### ISSUE THREAD

**Issue by sanchitprakash** [aler9/rtsp-simple-server#937]
## Which version are you using?

v0.18.2

## Which operating system are you using?

- [x] Linux amd64 standard
- [x] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue
Hello,

I'm using rtsp-simple-server for proxy streaming. For this, I dynamically change the configuration file by writing new entries in the _rtsp-simple-server.yml_ file. This _rtsp-simple-server.yml_ file is commonly mounted via Azure Fileshare to another python application that is making all the changes. For example:
paths:
  6144576EABCA401C9F7E7C63:
    source: rtsp://sourceIP:8554/machine_test
    sourceOnDemand: true
    sourceProtocol: tcp
  6144576EABCA401C9F7E7C63SUB:
    source: rtsp://sourceIP:8554/machine_test
    sourceOnDemand: true
    sourceProtocol: tcp
 
For the initial couple of entries, the proxies work fine and new changes are detected automatically without any issues. These newly added paths could also be listed if we curl http://localhost/v1/paths/list. However, after a certain number of entries (let's say 100) this auto-reloading of configuration stops working and it does not load the latest changes being made to the configuration file. Listing the paths curl http://localhost/v1/paths/list also does not show these new entries but, as soon we restart the rtsp-simple-server these new entered proxies are loaded properly. However, restarting the server every time a new entry is made is not feasible as it intrrupts the other active proxies which defy the hot reloading feature. 
As per the documentation:
"The configuration can be changed dinamically when the server is running (hot reloading) by writing to the configuration file. Changes are detected and applied without disconnecting existing clients, whenever it's possible."

## Describe how to replicate the issue

1. start the server
2. dynamically keep on adding new paths to the configuration file
3. server will fail to load the new proxy paths after a certain number of entries 

## Did you attach the server logs?

no

## Did you attach a network dump?

no

**Comment by aler9** [comment]
Hello,

First of all if you want to dynamically add paths i suggest you to use the [REST API](https://github.com/aler9/rtsp-simple-server#http-api) instead of hot reloading, since it's way more reliable.

Regarding the hot reloading feature, the problem is that the server reads the file before you've finished to write it, since the bigger the file, the more time it takes to be written to disk. You can overcome the issue by writing the config to a temporary file, i.e.

```
rtsp-simple-server.yml.tmp
```

and only after the writing is finished, replace the configuration file

```
mv rtsp-simple-server.yml.tmp rtsp-simple-server.yml
```

In this way, the server detects the update only after the writing is finished and reads the file thereafter.

**Comment by sanchitprakash** [comment]
Hello, 

Yes, I have been doing the same only .i.e overwrite the file with a temporary file and in addition with some timeout to ensure the file is written properly.
`mv rtsp-simple-server.yml.tmp rtsp-simple-server.yml`
However, this approach is not working as I have mounted the "rtsp-simple-server.yml" file using NFS Azure Fileshare and it's being edited by another application to add/remove paths dynamically. Even when I download the file from the Fileshare all the changes are reflected properly but the rtsp-simple-server fails to hot reload these changes.

Also, I cannot use RestAPI as I'm running rtsp-simple-server on Kubernetes as a pod that has multiple replicas masked by the same public IP and it would be difficult to track to which replica pod the request is being directed. Hence, I went ahead with the hot reloading feature expecting there would be a centralized configuration that I could edit dynamically and it would be reflected across all the replica pods of rtsp-simple-server (in the context of auto-scaling). Basically, if a streaming request came any of the replica pods could handle it as all the servers pods have a common configuration. I hope you got my point. Any suggestions would be really helpful.
Thanks
